client/state/receipts/assembler.js

```diff
@@ -12,8 +12,17 @@ export function createReceiptObject( data ) {
 				productName: purchase.product_name,
 				productNameShort: purchase.product_name_short,
 				registrarSupportUrl: purchase.registrar_support_url,
-				isEmailVerified: Boolean( purchase.is_email_verified )
+				isEmailVerified: Boolean( purchase.is_email_verified ),
 			};
-		} )
+		} ),
+		failedPurchases: data.failedPurchases && data.failedPurchases.map( purchase => {
```

Comment:
`failedPurchases: ( data.failedPurchases || [] ).map( purchase => {` ensures that the value will always be an array.